src/test/regress/expected/subquery_basics.out

```diff
@@ -503,3 +503,89 @@ DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT user_id FROM
+SELECT COUNT(*) FROM dist FULL JOIN ref USING(id)
+WHERE
+	dist.id IN (SELECT id FROM dist WHERE id > 5) AND
+    dist.value IN (SELECT value FROM dist WHERE id > 15);
+DEBUG:  generating subplan XXX_1 for subquery SELECT value FROM public.dist WHERE (id OPERATOR(pg_catalog.>) 15)
+DEBUG:  recursively planning left side of the full join since the other side is a recurring rel
+DEBUG:  recursively planning distributed relation "dist" since it is part of a distributed join node that is outer joined with a recurring rel
+DEBUG:  Wrapping relation "dist" to a subquery
+DEBUG:  generating subplan XXX_2 for subquery SELECT id, value FROM public.dist WHERE true
+DEBUG:  generating subplan XXX_3 for subquery SELECT id FROM public.dist WHERE (id OPERATOR(pg_catalog.>) 5)
+DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT count(*) AS count FROM ((SELECT dist_1.id, dist_1.value FROM (SELECT intermediate_result.id, intermediate_result.value FROM read_intermediate_result('XXX_2'::text, 'binary'::citus_copy_format) intermediate_result(id integer, value integer)) dist_1) dist FULL JOIN public.ref USING (id)) WHERE ((dist.id OPERATOR(pg_catalog.=) ANY (SELECT intermediate_result.id FROM read_intermediate_result('XXX_3'::text, 'binary'::citus_copy_format) intermediate_result(id integer))) AND (dist.value OPERATOR(pg_catalog.=) ANY (SELECT intermediate_result.value FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(value integer))))
+ count
+---------------------------------------------------------------------
+     0
+(1 row)
+
+-- subquery in WHERE clause should be recursively planned after planner recursively plans recurring full join
+SELECT id FROM ref FULL JOIN dist USING (id)
+WHERE
+	dist.id IN (SELECT id FROM dist GROUP BY id);
+DEBUG:  recursively planning right side of the full join since the other side is a recurring rel
+DEBUG:  recursively planning distributed relation "dist" since it is part of a distributed join node that is outer joined with a recurring rel
+DEBUG:  Wrapping relation "dist" to a subquery
+DEBUG:  generating subplan XXX_1 for subquery SELECT id FROM public.dist WHERE true
+DEBUG:  generating subplan XXX_2 for subquery SELECT id FROM public.dist GROUP BY id
+DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT id FROM (public.ref FULL JOIN (SELECT dist_1.id, NULL::integer AS value FROM (SELECT intermediate_result.id FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(id integer)) dist_1) dist USING (id)) WHERE (dist.id OPERATOR(pg_catalog.=) ANY (SELECT intermediate_result.id FROM read_intermediate_result('XXX_2'::text, 'binary'::citus_copy_format) intermediate_result(id integer)))
+ id
+---------------------------------------------------------------------
+(0 rows)
+
+-- subquery in WHERE clause should be recursively planned after planner recursively plans recurring full join
```

Comment:
isn't this the same query as the above?

Reply:
No. intermediate result columns are not matching in one of them.